source/proxy_strategy.c

```diff
@@ -561,6 +879,8 @@ struct aws_http_proxy_strategy_tunneling_chain {
     void *original_internal_proxy_user_data;
     aws_http_proxy_strategy_terminate_fn *original_strategy_termination_callback;
     aws_http_proxy_strategy_http_request_forward_fn *original_strategy_http_request_forward_callback;
+    enum proxy_strategy_connect_state connect_state;
```

Comment:
I don't think these are necessary.

Reply:
Removed connect_state from tunneling chain. This was not needed anymore after fixing the cast issue.